junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestNameFormatter.java

```
@@ -289,4 +303,34 @@ Set<String> placeholders() {
 		}
 	}
 
+	private static class DefaultNonPlaceholderFormatter implements NonPlaceholderFormatter {
+
+		private final PartialFormatter delegate;
+
+		public DefaultNonPlaceholderFormatter(String segment, int argumentMaxLength) {
+			this.delegate = segment.contains("{") //
+				? new MessageFormatPartialFormatter(segment, argumentMaxLength) //
+				: (context, result) -> result.append(segment);
+		}
+
+		@Override
+		public void append(ArgumentsContext context, StringBuffer result) {
+			delegate.append(context, result);
+		}
+	}
+
+	private static class ExpressionLanguageNonPlaceholderFormatter implements NonPlaceholderFormatter {
+
+		private final ExpressionLanguageAdapter expressionLanguageAdapter;
+
+		public ExpressionLanguageNonPlaceholderFormatter(ExpressionLanguageAdapter expressionLanguageAdapter) {
+			this.expressionLanguageAdapter = expressionLanguageAdapter;
+
+		}
+
+		@Override
+		public void append(ArgumentsContext argumentsContext, StringBuffer result) {
+			expressionLanguageAdapter.format(argumentsContext.arguments.get()[0], result);
```

This is not correct, but I'm unsure how we should name multiple arguments, so that they can be accessed by the EL?

---

Regarding the `Map<String, Object>`, that's also what I suggested in https://github.com/junit-team/junit5/issues/1154#issuecomment-345445625.

> In summary, I believe it would be better to design a new extension API that receives the `String displayName` and a `Map<String,Object> expressionContextMap` and then allow people to pick between Mustache, SpEL, etc.

---

> That would require compiling with `javac`'s `-parameters` option. We already check for `parameter.isNamePresent()` in `ParameterizedTestMethodContext.getParameterName(int)`, though, so we could populate the `Map` based on that and fail (?) if no parameter names are present?

I don't think I would throw an exception. Rather, I would document the behavior, pointing out that source code parameter names are available when compiled with `-parameters` and that otherwise the arguments are available via `arg0`, `arg1`, etc. (i.e., the default parameter names returned by `Parameter.getName()`).